src/main/java/com/bravo/user/controller/UserController.java

```diff
@@ -43,10 +43,33 @@ public UserReadDto create(final @RequestBody UserSaveDto request, final BindingR
     return userService.create(request);
   }
 
-  @GetMapping(value = "/retrieve/{id}")
+  @GetMapping(value = "/retrieveId/{id}")
   @ResponseBody
-  public UserReadDto retrieve(@PathVariable String id){
-    return userService.retrieve(id);
+  public UserReadDto retrieveById(@PathVariable String id){
+    return userService.retrieveById(id);
+  }
+  
+  /**
+   * Retrieves a paginated list of users whose first, middle, and last names match the specified name.
+   * The name to search may contain negation ("!") or wildcard ("*" or "%") characters
+   * 
+   * @param name Name to find - should contain first, middle, and last name without space
+   * @param page Page number to retrieve
+   * @param size Size of the page to retrieve
+   * @param response {@linkplain HttpServletResponse}
+   * @return List of {@linkplain UserReadDto}s with combined first, middle, and last names that match the search criteria
+   */
+  @GetMapping(value = "/retrieveName/{name}")
```

Comment:
I wouldn't expect uppercase in the URL. Perhaps something like:
* /retrieve/id/{id} and /retrieve/name/{name}

I would've preferred a common GET method that took the values as optional RequestParam's
* /retrieve?id={id}
* /retrieve?name={name}

Reply:
Yep, I get that.  I wasn't sure if it was acceptable to modify the existing code to work with RequestParams instead of PathVariables or just add an additional method, so I went with the latter.  Will update though.